internal/framework/resources/registered_domain/create.go

```diff
@@ -0,0 +1,194 @@
+package registered_domain
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/dnsimple/dnsimple-go/dnsimple"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/terraform-providers/terraform-provider-dnsimple/internal/consts"
+	"github.com/terraform-providers/terraform-provider-dnsimple/internal/framework/utils"
+)
+
+func (r *RegisteredDomainResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *RegisteredDomainResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	domainAttributes := dnsimple.RegisterDomainInput{
+		RegistrantID: int(data.ContactId.ValueInt64()),
+	}
+
+	if !data.AutoRenewEnabled.IsNull() {
+		domainAttributes.EnableAutoRenewal = data.AutoRenewEnabled.ValueBool()
+	}
+
+	if !data.WhoisPrivacyEnabled.IsNull() {
+		domainAttributes.EnableWhoisPrivacy = data.WhoisPrivacyEnabled.ValueBool()
+	}
+
+	if !data.PremiumPrice.IsNull() {
+		domainAttributes.PremiumPrice = data.PremiumPrice.ValueString()
+	}
+
+	if !data.ExtendedAttributes.IsNull() {
+		extendedAttrs := make(map[string]string)
+		diags := data.ExtendedAttributes.ElementsAs(ctx, &extendedAttrs, false)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		domainAttributes.ExtendedAttributes = extendedAttrs
+	}
+
+	lowerCaseDomainName := strings.ToLower(data.Name.ValueString())
```

Comment:
We should just validate the domain name here and bail if it's not lowercase. Otherwise:

- User may be surprised that domain is actually different.
- Potential Terraform state conflict/drift issues.
- We have to ensure we do this everywhere when calling our API.

Reply:
Validator already covers this.